test/Interpreter/SDK/check_class_for_archiving.swift

```diff
@@ -0,0 +1,51 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift %s -module-name=_Test -import-objc-header %S/Inputs/check_class_for_archiving.h -o %t/a.out
+// RUN: %target-run %t/a.out | %FileCheck %s
+
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
+// REQUIRES: CPU=i386_or_x86_64
+// UNSUPPORTED: OS=tvos
+// UNSUPPORTED: OS=watchos
```

Comment:
I think we should be able to handle any Apple platform. The first two lines should be sufficient.